internal/driver/common/canvas.go

```diff
@@ -314,8 +314,10 @@ func (c *Canvas) Painter() gl.Painter {
 
 // Refresh refreshes a canvas object.
 func (c *Canvas) Refresh(obj fyne.CanvasObject) {
-	c.refreshQueue.In(obj)
-	c.SetDirty()
+	async.EnsureMain(func() {
+		c.refreshQueue.In(obj)
+		c.SetDirty()
+	})
```

Comment:
Suggestion but not request: It seems unnecessary to do move the refreshQueue.In() call to another thread as it is an async channel anyway. Maybe this is cleaner?
```suggestion
        c.refreshQueue.In(obj)
	async.EnsureMain(c.SetDirty)
```

Reply:
Sorry that the indenting was off. Apparently GitHub doesn't handle that very well. Looked good on my computer